pandas/core/internals/blocks.py

```diff
@@ -944,15 +944,14 @@ def putmask(self, mask, new, align=True, inplace=False, axis=0, transpose=False)
                 and np.any(mask[mask])
                 and getattr(new, "ndim", 1) == 1
             ):
-
-                if not (
-                    mask.shape[-1] == len(new)
-                    or mask[mask].shape[-1] == len(new)
-                    or len(new) == 1
-                ):
+                if mask[mask].shape[-1] == len(new):  # GH 30567
+                    np.place(new_values, mask, new)
```

Comment:
i had to look up what np.place was.  is there a more common idiom we can use here?

Reply:
I can't find any close alternative from numpy. I think `np.place` in the right choice as it does exactly what we need here (it places all given values to the masked locations rather than assigning masked values to masked locations). Besides, it is designed to specifically work with book masks and that's what we are dealing with here.